### PR TITLE
Replace MPRIS implementation

### PR DESCRIPTION
**Pull request by suransea**
I created the `dbus-swift` and `mpris-swift` libraries, providing a new `MPRIS` implementation. The new implementation eliminates external dependencies and adds support for `macOS`.

Changes: 
- Removed the old MPRIS implementation.
- Integrated a new MPRIS implementation for better compatibility.
- Updated relevant code and dependencies to support the new implementation.

### CODE REVIEW

**Comment by suransea** [comment]
The CI build is currently failing because Swift 6 is not available.

File: .clang-format
Comment (ddddxxx) [inline .clang-format:1]: do not commit unrelated files (also gitignore).

File: Sources/MusicPlayer/Players/MPRIS.swift [hunk 1]
```diff
@@ -7,145 +7,110 @@
...
-            signals.append(
-                g_signal_connect_data(player, "seeked", unsafeBitCast(onSeeked, to: GCallback?.self), pself, nil, G_CONNECT_AFTER)
-            )
-            signals.append(
-                g_signal_connect_data(player, "metadata", unsafeBitCast(onMetadataChanged, to: GCallback?.self), pself, nil, G_CONNECT_AFTER)
-            )
+            let updatePlayerState: () -> Void = { [weak self] in self?.updatePlayerState() }
+            disposables.append(try player.player.playbackStatus.observe(updatePlayerState))
+            disposables.append(try player.player.metadata.observe(updatePlayerState))
+            disposables.append(try player.player.seeked { _ in updatePlayerState() })
```
Comment (ddddxxx) [inline Sources/MusicPlayer/Players/MPRIS.swift:28]: [weak self]

File: Sources/MusicPlayer/Players/MPRIS.swift [hunk 2]
```diff
@@ -7,145 +7,110 @@
...
             updatePlayerState()
         }
-        
+
         deinit {
-            for var signal in signals {
-                g_clear_signal_handler(&signal, player)
-            }
-            g_object_unref(player)
+            disposables.forEach { try? $0() }
```
Comment (ddddxxx) [inline Sources/MusicPlayer/Players/MPRIS.swift:33]: The name is misleading, we can't just dispose of them, right?

Can you wrap them in DBus library, like this?

```swift
class DBusHandlerRegistation {
    typealias UnregisterFuction = () throws(DBus.Error) -> Void
    let unregister: UnregisterFuction
    deinit {
        try? unregister()
    }
}
```

File: Sources/MusicPlayer/Players/MPRIS.swift [hunk 3]
```diff
@@ -7,145 +7,110 @@
...
+    ) throws {
+        let player = MediaPlayer2(
+            connection: connection, name: name, instance: instance, timeout: timeout)
+        try self.init(player: player)
     }
 }
 
 extension MusicPlayers.MPRIS: MusicPlayerProtocol {
-    
+    public var name: MusicPlayerName? { nil }
```
Comment (ddddxxx) [inline Sources/MusicPlayer/Players/MPRIS.swift:71]: use bus name? or just `.mpris` if not feasible.

File: Sources/MusicPlayer/Players/MPRIS.swift [hunk 4]
```diff
@@ -7,145 +7,110 @@
...
-        g_list_free(playerNames)
-        return result
+    /// Initializes a new MPRIS music player.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the media player.
+    ///   - instance: The instance name of the media player, if any.
+    ///   - queue: The dispatch queue for the D-Bus method calls.
+    public convenience init(
+        name: String, instance: String? = nil, queue: DispatchQueue? = nil
```
Comment (ddddxxx) [inline Sources/MusicPlayer/Players/MPRIS.swift:46]: I think it's better to make it as simple as possible, treat queue, connection, and timeout as implementation details and don't expose them. We can always expose them later if someone needs it.

File: Sources/MusicPlayer/Players/MPRISNowPlaying.swift [hunk 5]
```diff
@@ -7,79 +7,54 @@
...
-    public final class MPRISNowPlaying: NowPlaying {
-        
-        private let manager: UnsafeMutablePointer<PlayerctlPlayerManager>
-        private var signals: [gulong] = []
-        
-        public init?() {
-            guard let manager = playerctl_player_manager_new(nil) else {
-                return nil
+    /// Just like `NowPlaying`, but automatically find available MPRIS players.
+    public final class MPRISNowPlaying: Agent {
```
Comment (ddddxxx) [inline Sources/MusicPlayer/Players/MPRISNowPlaying.swift:16]: Here we have player selection logic twice, in different repo. How about deriving from `NowPlaying`, only update player list and let `NowPlaying` do the selection?

**Review by ddddxxx — CHANGES_REQUESTED** [review]
It's been a while :). The new MPRIS implementation looks brilliant. Here's my suggestion. Thank you!

**Pull request closed without merging**